mautic/mautic
# DPMMA-1436 click-through stats in email report

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ N ]
| New feature/enhancement? (use the a.x branch)      | [ Y ]
| Deprecations?                          | [ N ]
| BC breaks? (use the c.x branch)        | [ N ]
| Automated tests included?              | [ Y ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
This PR adds new columns to Emails  reports:
- Click-through count
- Click-through rate (CTR)
- Click-to-open rate (CTOR)

![image](https://github.com/mautic/mautic/assets/8580942/2b920ad6-b477-4e11-96cd-7934aeb0a010)



#### Steps to test this PR:



1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a new email with links and send to any segment with contacts
3. Generate some clicks (read and click links as contact)
4. Create a report with datasource **Emails** and add columns: Click-through count, Click-through rate, Click-to-open rate
5. Save & close and check if values are correct
